submissions/Harnytskyi/friends-app/js/script.js

```diff
@@ -0,0 +1,163 @@
+const GRID = document.getElementById("allFriends");
+const CHECKBOX_MAN = document.getElementById("gender-male");
+const CHECKBOX_WOMAN = document.getElementById("gender-female");
+const SORT_ALPHABET_ASCEND = document.getElementById("a-z");
+const SORT_ALPHABET_DESCEND = document.getElementById("z-a");
+const SEARCH_INPUT = document.getElementById('search');
+const SORT_AGE_ASCEND = document.getElementById("0-9");
+const SORT_AGE_DESCEND = document.getElementById("9-0");
+const FILTER_SORT = document.getElementById("filter-sort");
+const FILTER_GENDER = document.getElementById("filter-gender");
+let friends = [];
+let gender = {
+    male: true,
+    female: true
+}
+
+let selectedFriends = [...friends];
+function filterByGender() {
+    if (gender.male == true && gender.female == true)
+        selectedFriends = friends;
+    else if (gender.male == true && gender.female == false)
+        selectedFriends = friends.filter(item => item.gender == "male")
+    else if (gender.male == false && gender.female == true)
+        selectedFriends = friends.filter(item => item.gender == "female")
+    search();
```

Comment:
Same as with your `sortFunction`. You should not call search from `filterByGender` it creates unpredictable side effects.

You should move this call outside of this function.